### ISSUE TITLE
cat function in hw#3

### ISSUE THREAD

**Issue by rgosh001** [mikeizbicki/ucr-cs100#82]
Just because it crossed my mind, are we suppose to call cat in execv (if that's even possible to)
or do we just hardcode it with the open function and change stdout/stdin to the file itself?

**Comment by mikeizbicki** [comment]
I don't understand your question, can you rephrase it and be more specific?

**Comment by rgosh001** [comment]
I'm sorry. I'll try again.
When we're dealing with file-redirection (for example "cat < file1"), are we able to use execv to call cat or do we need to implement that feature ourself with open(file1, o_rdonly) and manipulate stdin/stdout? In simpler terms, is calling cat with execv even possible?

**Comment by mikeizbicki** [comment]
You MUST use `exec`.  The program you are doing redirection on could be any possible program, not just `cat`.

**Comment by rgosh001** [comment]
Understood
Just to clarify even deeper, if I had `cat < file1` , what would I pass into execv besides "cat" as the first argument? but before I even call execv on `cat`,  I would also need to open the file, read, and write to either stdin/stdout. Am I correct?

**Comment by mikeizbicki** [comment]
Correct.  You would not be passing any arguments to `cat`.

**Comment by rgosh001** [comment]
Ahh understood. I will try it again and see my results.
